log4j-core/src/main/java/org/apache/logging/log4j/core/appender/SmtpAppender.java

```diff
@@ -261,9 +277,21 @@ public SmtpAppender build() {
             if (getFilter() == null) {
                 setFilter(ThresholdFilter.createFilter(null, null, null));
             }
-            final SmtpManager smtpManager = SmtpManager.getSmtpManager(getConfiguration(), to, cc, bcc, from, replyTo,
-                    subject, smtpProtocol, smtpHost, smtpPort, smtpUsername, smtpPassword, smtpDebug,
-                    getFilter().toString(), bufferSize, sslConfiguration);
+            if (Strings.isEmpty(smtpProtocol)) {
+                smtpProtocol = "smtp";
+            }
+            final Serializer subjectSerializer = PatternLayout.newSerializerBuilder()
+                    .setConfiguration(getConfiguration())
+                    .setPattern(subject)
+                    .build();
+            final FactoryData data = new FactoryData(to, cc, bcc, from, replyTo, subject, subjectSerializer,
+                    smtpProtocol, smtpHost, smtpPort, smtpUsername, smtpPassword, smtpDebug, bufferSize,
+                    sslConfiguration, getFilter().toString());
+            final MailManager smtpManager = AbstractManager.getManager(data.getManagerName(), FACTORY, data);
+            if (smtpManager == null) {
+                LOGGER.error("Unabled to instantiate SmtpAppender named {}", getName());
```

Comment:
Should `return null` here as well to avoid further invalid state when constructing the appender below.